### PR TITLE
perf: use derived atoms for solar and wind state

### PR DESCRIPTION
**Pull request by VIKTORVAV99**
## Description

Uses derived atoms for the wind and solar enabled state to reduce code duplication and minimise unneeded imports.

### Double check
- [x] I have run `pnpx prettier@2 --write .` and `poetry run format` in the top level directory to format my changes.

### CODE REVIEW

File: web/src/utils/state/atoms.ts [hunk 1]
```diff
@@ -29,8 +29,19 @@ export const spatialAggregateAtom = atomWithStorage(
 );
 export const productionConsumptionAtom = atomWithStorage('mode', Mode.CONSUMPTION);
 
-export const solarLayerEnabledAtom = atomWithStorage('solar', ToggleOptions.OFF);
+export const solarLayerAtom = atomWithStorage('solar', ToggleOptions.OFF);
+export const isSolarLayerEnabledAtom = atom(
+  (get) =>
+    get(solarLayerAtom) === ToggleOptions.ON &&
+    get(selectedDatetimeIndexAtom).index === 24
```
Comment (madsnedergaard) [inline web/src/utils/state/atoms.ts:36]: We need to add a check for `isHourly` (from #6980), as it is currently possible to make the solar layer show on daily view (on day 25) 😬 

Screenshot 2024-07-16 at 14 40 24

**Review by madsnedergaard — COMMENTED** [review]
LGTM, but reading the existing code made me realise we have a silly bug in here :)

File: web/src/utils/state/atoms.ts (hunk 1 above)
Comment (VIKTORVAV99) [reply web/src/utils/state/atoms.ts:38]: Should be fixed now

**Review by madsnedergaard — APPROVED** [review]
Thanks for fixing the bug :)

**Pull request merged**